### ISSUE TITLE
"Vagrant up" hangs on windows 7. Due to powershell?

### ISSUE THREAD

**Issue by daffy1234** [mitchellh/vagrant#8783]
Please note that the Vagrant issue tracker is reserved for bug reports and
enhancements. For general usage questions, please use the Vagrant mailing list:
https://groups.google.com/forum/#!forum/vagrant-up. Thank you!

### Vagrant version
Vagrant 1.9.7

### Host operating system
Windows 7 Professional

### Guest operating system
Ubuntu Server 14.04 ("ubuntu/trusty64" box)

### Vagrantfile
Comments were removed.

```ruby
$infoScript = <<SCRIPT
  echo 'IP-addresses of the vm ...'
  ifconfig
SCRIPT

Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/trusty64"
  config.vm.network "public_network"
  config.vm.synced_folder "www", "/var/www"
  config.vm.synced_folder "phpconfig", "/etc/php5/apache2"
  config.vm.provision "shell", inline: <<-SHELL
    apt-get update
    apt-get install -y apache2 php5 libapache2-mod-php5
	sudo /etc/init.d/apache2 restart
  SHELL
  config.vm.provision "shell", inline: $infoScript,
      run: "always"
end
```

### Debug output
https://gist.github.com/daffy1234/373f51a56ba183b886d045da99643011

### Expected behavior
The machine should be provisioned and come up successfully.

### Actual behavior
After the initial "vagrant up" command, absolutely no output is produced. vagrant.exe seems to open ruby.exe and powershell.exe. Killing powershell.exe twice is a workaround, and it gets the machine provisioned and booting normally. This issue does not occur after the machine has successfully booted at least once.

### Steps to reproduce
1. "vagrant up" on a brand new machine.

**Comment by daffy1234** [comment]
Downgrading to Vagrant 1.9.6 is another workaround.

**Comment by ikus060** [comment]
I've hit the same issue on a fresh Windows 7 install with Vagrant 1.9.7.

**Comment by marcuslang** [comment]
Same here. It hangs here:

> INFO subprocess: Starting process: ["C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\/powershell.EXE", "-NoLogo", "-NoProfile", "-NonInteractive", "-
ExecutionPolicy", "Bypass", "-Command", "(new-object System.Security.Principal.WindowsPrincipal([System.Security.Principal.WindowsIdentity]::GetCurren
t())).IsInRole([System.Security.Principal.WindowsBuiltInRole]::Administrator)"]
INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: True

**Comment by ferrarimarco** [comment]
Same issue on Windows 7 x64, even running `vagrant status`

**Comment by bbkane** [comment]
I have this issue as well.

**Comment by markehb** [comment]
Same here, going back to 1.9.6 resolved for me as well.

**Comment by rashidkukkady** [comment]
I have similar issue... have not tried vagrant 1.9.6 as yet. Will do and update

**Comment by sd1ver** [comment]
I've got the same issue.
The last message with --debug key is like the posted above:
DEBUG subprocess: stdout: False

**Comment by badhop** [comment]
Have the same problem.

**Comment by edwardmlyte** [comment]
As has been commented on https://github.com/mitchellh/vagrant/issues/8777, killing the powershell.exe process allows vagrant to continue, should you really NEED the latest version for some other reason.

**Comment by bozemojmaciej** [comment]
Hello, same issue appeared today. Fix that worked for me was to update Windows PowerShell (was 2009). Now it is 2016 version and Vagrant works perfect. You can update it by installing WMF: https://msdn.microsoft.com/en-us/powershell/mt173057.aspx

**Comment by pscheit** [comment]
upgrading powershell helps (install WMF 5)
you commented half an hour earlier :D

**Comment by syagawa** [comment]
Update Windows Powershell to ver 5, the problem was solved.

Vagrant: Vagrant 1.9.7
Host OS: Windows 7 Professional 64bit
Guest OS: Ubuntu Server 14.04 ("ubuntu/trusty64" box)

**Comment by ikus060** [comment]
Upgrading powershell cannot be a final fix for this bug. I recommend to show a warning in vagrant, asking the user to upgrade power shell...

**Comment by dfad1ripe** [comment]
WMF 5.1 + .NET 4.6.1 works for me too on Win 7 SP1. Even with that, after launching 'vagrant up', it freezes for ~20 sec, while powershell.exe eats up to 50% of i5 CPU resources. I don't remember such behavior on earlier versions.

**Comment by mikeyo241** [comment]
I had the same problem running vagrant up for "ubuntu/trusty64" it would hang on:
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: True.  
I installed WMF like @bozemojmaciej suggested and the problem is fixed!!

**Comment by ashokvaddevalli** [comment]
I am also getting same issue on using Vagrant 1.9.6 version.

INFO subprocess: Starting process: ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\/powershell.EXE", "-NoLogo", "-NoProfile", "-NonInteractive", "-ExecutionPolicy", "Bypass", "-Command", "(new-object System.Security.Principal.WindowsPrincipal([System.Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([System.Security.Principal.WindowsBuiltInRole]::Administrator)"]
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: True

**Comment by zafaralam** [comment]
Had the same issue on my windows 7 machine.

Updated Windows Poweshell to version 5 and all works now.

Thanks a lot.

**Comment by ashokvaddevalli** [comment]
Updating Powershell on Host Machine or guest VM?

**Comment by pscheit** [comment]
host

**Comment by ashokvaddevalli** [comment]
Thanks @pscheit 

I have installed Powershell on host machine(Vagrant installed machine), now it was working now. 

But I am getting another issue,

 INFO synced_folders: Synced Folder Implementation: smb
 INFO synced_folders:   - /vagrant: . => /vagrant
 INFO synced_folders: Invoking synced folder prepare for: smb
 INFO interface: output: Preparing SMB shared folders...
 INFO interface: output: ==> default: Preparing SMB shared folders...
 INFO interface: detail: You will be asked for the username and password to use for the SMB
folders shortly. Please use the proper username/password of your
Windows account.

 INFO interface: detail:     default: You will be asked for the username and password to use for the SMB
    default: folders shortly. Please use the proper username/password of your
    default: Windows account.
    default:
 INFO interface: ask: Username:
 INFO interface: ask:     default: Username:

**Comment by wiltaylor** [comment]
Prompting you for Samba folder credentials is normal. It is asking for an account on your host to connect to. You can put the credentials in the vagrant file but it isn't very secure to do so.

One work around I have is I store credentials in some environment variables and pass those into the vagrantfile. I set the environment variables when I open powershell to help keep them only to my vagrant session.

Example:

```
Vagrant.configure("2") do |config|
  config.vm.box = "Windows2012R2"
  config.vm.network "public_network", :bridge => "NAT
  config.vm.synced_folder ".", "/vagrant", type: "smb", smb_username: ENV["smb_username"], smb_password: ENV["smb_password"]
  end
end
```

**Comment by ashokvaddevalli** [comment]
Thanks @wiltaylor 

I have manually entered guest VM username and password, and now I am getting the below error.

As another option, you can manually specify an IP for the machine
to mount from using the `smb_host` option to the synced folder.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: We couldn't detect an IP address that was routable to this
machine from the guest machine! Please verify networking is properly
setup in the guest machine and that it is able to access this
host.

 INFO interface: Machine: error-exit ["VagrantPlugins::VSphere::Errors::VSphereError", "We couldn't detect an IP address that was routable to this\nmachine from the guest machine! Please verify networking is properly\nsetup in the guest machine and that it is able to access this\nhost.\n\nAs another
 option, you can manually specify an IP for the machine\nto mount from using the `smb_host` option to the synced folder."]

**Comment by ashokvaddevalli** [comment]
Thanks @wiltaylor 

Can I use credentials for Host machine or guest VM?

**Comment by Bilge** [comment]
Confirm same hanging issue on Win 7 x64 with Vagrant 1.9.7.

**Comment by ashokvaddevalli** [comment]
I am unable to access c:/vagrant folder on Guest VM, I am getting below error.

Pop up window: Location is not available
 C:\vagrant is not accessible
 Logon failure: unknown username or bad password.

**Comment by maheshbisen** [comment]
Its working for me after updating windows powershell.

Vagrant: Vagrant 1.9.7
Host OS: Windows 7 Professional 32bit
Guest OS: Ubuntu Server 14.04 ("ubuntu/trusty32" box)

**Comment by ashokvaddevalli** [comment]
Thanks @wiltaylor 

My issue is resolved.

**Comment by yassineselmi** [comment]
Cool! It is works after updating to powershell 5.0 .
Thank you.

**Comment by kylerdmoore** [comment]
Upgrading PowerShell fixed for me. Was running PowerShell 2. Upgraded to PowerShell 4 with the Windows Management Framework (WMF) 4.0 and vagrant no longer got hung up at "vagrant up" command.

Vagrant 1.9.8
Windows 7 Professional 64bit
VirtualBox 5.1.26
Using trusty64 as host vm

Here's the guide I used to upgrade PowerShell. DOTNET framework in this post is outdated but the rest is correct. [http://mikefrobbins.com/2015/01/08/how-to-check-the-powershell-version-and-install-a-new-version/](url)

**Comment by chrisroberts** [comment]
Upgrading powershell should resolve the issue. The latest release (1.9.8) includes this PR #8839 that checks the powershell version and raises an error if the minimum is not satisfied. Cheers!

**Comment by MorganGeek** [comment]
@chrisroberts even on 1.9.8 I still need to kill powershell.exe in order to see the warning...

**Comment by elconde** [comment]
@chrisroberts On 2.0.0 I also need to kill powershell.exe to see this warning.

**Comment by elconde** [comment]
@kylerdmoore The link in your post is broken.

http://mikefrobbins.com/2015/01/08/how-to-check-the-powershell-version-and-install-a-new-version/

**Comment by hummal** [comment]
@chrisroberts I have 2.0.0 installed and I don't see any warning. Not even if I kill the process. Updating PS to v5 helps

**Comment by timmjd** [comment]
Last working version for Windows 7 seems to be 1.9.5. Haven't tested 1.9.6 yet, but its broken since 1.9.7.

**Comment by StalkAlex** [comment]
@chrisroberts your PR is not doing what you described. There wasn't any exceptions/warnings on last version 2.0.0, windows 7, getting here is what help me to solve the problem.

**Comment by dmz9** [comment]
here is a faster and easier way of applying the workaround about installing new version of powershell.
do at your own risk, worked for me (not having any critical powershell-based startup scripts)

1. install separate powershell 6 beta (atm). here is the git repo https://github.com/PowerShell/PowerShell/releases
for win7 x64 i picked .msi version
2. by default it gets installed to C:\Program Files\PowerShell\6.0.0-beta.8 (my version)
3. rename old C:\Windows\System32\WindowsPowerShell\v1.0 to _v1.0 (or whatever) to have rollback just in case
4. copy C:\Program Files\PowerShell\6.0.0-beta.8\* into newly created C:\Windows\System32\WindowsPowerShell\v1.0\*
5. reboot (optional)
6. profit

vagrant loaded ok

**Comment by anthony-o** [comment]
With latest [v6.0.0-rc](https://github.com/PowerShell/PowerShell/releases/tag/v6.0.0-rc), @dmz9 technic failed: `vagrant up` ends in error with:
```
Failed to locate the powershell executable on the available PATH. Please
ensure powershell is installed and available on the local PATH, then
run the command again.
```
And indeed, there is no `powershell.exe` but a `pwsh.exe` in  `C:\Program Files\PowerShell\6.0.0-rc`. If I duplicate `pwsh.exe` to `powershell.exe` in `C:\Windows\System32\WindowsPowerShell\v1.0` after step 4, then `vagrant up` fails with:
```
The version of powershell currently installed on this host is less than
the required minimum version. Please upgrade the installed version of
powershell to the minimum required version and run the command again.

  Installed version: N/A
  Minimum required version: 3
```

**Comment by jcarey987** [comment]
So i updated powershell, and i type vagrant ssh into my terminal, which logs me into my machine. while i am logged in, the command prompt disappears. anyone else seeing this?

**Comment by robertluwang** [comment]
It is very interesting, vagrant up just hanging there forever, only saw exception when I killed all powershell process, 
```
ERROR vagrant: #<Vagrant::Errors::PowerShellInvalidVersion: The version of powershell currently installed on this host is less than
the required minimum version. Please upgrade the installed version of
powershell to the minimum required version and run the command again.

  Installed version: N/A
  Minimum required version: 3>
```
checked my powershell is 2, 
```
$PSVersionTable.PSVersion
```
Solved the issue by upgraded WMF to 4.0.

**Comment by MlleDelphine** [comment]
I am facing the same issue now.
Today, I've up my Vagrant VMs.
I've halted them.
I've destroyed one VM and try a "vagrant up" to recreate.

```
Failed to locate the powershell executable on the available PATH. Please
ensure powershell is installed and available on the local PATH, then
run the command again.
```
**Vagrant 2.0.1**
**Windows 10 Pro, 64bits**
**PowerShell verson 
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      16299  98**

**Comment by navigatrum** [comment]
Same problem.
Updated from Vagrant 1.8.5 to Vagrant 2, new Vagrantfile worked, but on vagrant up:

> The version of powershell currently installed on this host is less than
> the required minimum version. Please upgrade the installed version of
> powershell to the minimum required version and run the command again.

```
  Installed version: N/A
  Minimum required version: 3
```

Vagrant Version 2.0.3
Windows version: 10 Home, 64bits
PowerShell version: 5.1.16299.251

**Comment by briancain** [comment]
Hi @navigatrum - this issue has already been reported in #9629. Thanks!

**Comment by ultrasamad** [comment]
Restarting the pc in case of windows after adding the box before running `vagrant up` did a fix for me.

**Comment by alhaponyfarag** [comment]
In my case, this issue was solved by updating powershell and .Net framework
